Firmware/Marlin_main.cpp

```diff
@@ -3843,31 +3825,23 @@ static void gcode_M600(bool automatic, float x_position, float y_position, float
     custom_message_type = CustomMsg::Status;
 }
 
-void gcode_M701(uint8_t mmuSlotIndex){
-    printf_P(PSTR("gcode_M701 begin\n"));
-    
+void gcode_M701(float fastLoadLength, uint8_t mmuSlotIndex){
     FSensorBlockRunout fsBlockRunout;
     
     if (farm_mode) {
         prusa_statistics(22);
     }
 
     if (MMU2::mmu2.Enabled() && mmuSlotIndex < MMU_FILAMENT_COUNT) {
-        MMU2::mmu2.load_filament(mmuSlotIndex); // loads current extruder
-        // mmu_extruder = mmuSlotIndex; // @@TODO shall load filament set current tool to some specific index? We don't do that anymore.
+        MMU2::mmu2.load_filament_to_nozzle(mmuSlotIndex);
     } else {
-        enable_z();
         custom_message_type = CustomMsg::FilamentLoading;
 
         lcd_setstatuspgm(_T(MSG_LOADING_FILAMENT));
-        current_position[E_AXIS] += 40;
+        current_position[E_AXIS] += fastLoadLength;
         plan_buffer_line_curposXYZE(400 / 60); // fast sequence
         st_synchronize();
 
```

Comment:
why was the Z-move removed?

Reply:
Yes it is now done in the M701 gcode with `raise_z`. Now it applies to both single material and MMU.